### PR TITLE
TST: Fix issue with test order dependence

### PR DESCRIPTION
**Pull request by bashtage**
Fix test order dependence in test_sarimax.Test_ar_no_enforce
where changing order of tests could result in unexpected failure

### CODE REVIEW

**Comment by bashtage** [comment]
@ChadFulton test code modified the model and so the order that the individual tests run matters.  Do my changes still make the test meaningful?

**Comment by ChadFulton** [comment]
Thank you for looking into and fixing this. I think the test is still valid. I just had one question which I'll put inline.

File: statsmodels/tsa/statespace/tests/test_sarimax.py [hunk 1]
```diff
@@ -1170,16 +1167,23 @@ def setup_class(cls, *args, **kwargs):
         # of states if enforce_stationarity = False
         cls.model.ssm.loglikelihood_burn = 0
 
-    def test_loglike(self):
```
Comment (ChadFulton) [inline statsmodels/tsa/statespace/tests/test_sarimax.py]: Was this test causing problems? I can't tell why we needed to eliminate it.

Comment (bashtage) [inline statsmodels/tsa/statespace/tests/test_sarimax.py]: This test seems to pass fine using the version in the parent class (pasted below).  It seem that the reduced tolerance wasn't necessary, and so this is just a little clean up.  The `init_keys_replication` needed some fixed to pass.  Not sure if these are strictly kosher or not -- "fixing the test" might be proving that it doesn't work as expected.

```
    def test_loglike(self):
        self.result = self.model.filter(self.true_params)

        assert_allclose(
            self.result.llf,
            self.true_loglike,
            atol=0.7 * 10**(-self.decimal)
        )
```

**Comment by josef-pkt** [comment]
Is the ready for merging?

**Comment by bashtage** [comment]
I'm happy with it.  It doesn't change any tests that are run, just makes
them order invariant.

**Comment by josef-pkt** [comment]
Good, merging

Thanks

**Pull request merged**